### PR TITLE
ENH: scipy.interpolate: add derivative/antiderivative and inverse interpolation examples

### PR DESCRIPTION
**Pull request by mdhaber**
Addresses most elements of gh-730. I don't think I'll add an example of finding roots because that is partially addressed with the inverse interpolation example here, and the `roots` method is only available for `PPoly` splines, which we're not showing in this example already.

### CODE REVIEW

File: intro/scipy/index.rst [hunk 1]
```diff
@@ -342,6 +342,33 @@ exactly through each point.
...
+    >>> d_interp_results = d_interp_spline(interpolation_time)
+    >>> i_interp_spline = lambda t: interp_spline.antiderivative()(t) - 1
+    >>> i_interp_results = i_interp_spline(interpolation_time)
+
+.. image:: auto_examples/images/sphx_glr_plot_interpolation_003.png
+    :target: auto_examples/plot_interpolation.html
+    :scale: 60
+    :align: right
+
+For functions that are monotonic on an interval (e.g. :math:`sin` from :math:`\pi/2`
```
Comment (mdhaber) [inline intro/scipy/index.rst:360]:
```suggestion
For functions that are monotonic on an interval (e.g. :math:`\sin` from :math:`\pi/2`
```

File: intro/scipy/index.rst [hunk 2]
```diff
@@ -342,6 +342,33 @@ exactly through each point.
...
+to :math:`3\pi/2`), we can reverse the arguments of ``make_interp_spline`` to
+interpolate the inverse function. Because the first argument is expected to be
+monotonically *increasing*, we also reverse the order of elements in the arrays
+with `numpy.flip`.
```
Comment (mdhaber) [inline intro/scipy/index.rst:364]: Intersphinx link not working. Ideas?

**Review by mdhaber — COMMENTED** [review]
Other than these comments (and the fact that all math on the page is comically large), I think this rendered OK.
https://output.circle-artifacts.com/output/job/79699f22-d856-41bf-8078-c1f4c055c210/artifacts/0/build/html/intro/scipy/index.html#interpolation-scipy-interpolate

**Review by jarrodmillman — APPROVED** [review]

File: intro/scipy/index.rst [hunk 3]
```diff
@@ -342,6 +342,33 @@ exactly through each point.
...
+.. image:: auto_examples/images/sphx_glr_plot_interpolation_003.png
+    :target: auto_examples/plot_interpolation.html
+    :scale: 60
+    :align: right
+
+For functions that are monotonic on an interval (e.g. :math:`\sin` from :math:`\pi/2`
+to :math:`3\pi/2`), we can reverse the arguments of ``make_interp_spline`` to
+interpolate the inverse function. Because the first argument is expected to be
+monotonically *increasing*, we also reverse the order of elements in the arrays
+with `numpy.flip`.
```
Comment (jarrodmillman) [inline intro/scipy/index.rst:364]:
```suggestion
with :func:`numpy.flip`.
```

File: intro/scipy/index.rst (hunk 2 above)
Comment (mdhaber) [reply intro/scipy/index.rst]: Got it. I'll try to remember to be explicit about roles.

**Pull request merged**